components/domains/auth/SigninForm.tsx

```diff
@@ -0,0 +1,118 @@
+import React from "react";
+import styled from "styled-components";
+import SubmitButton from "./SubmitButton";
+import {
+  InputBox,
+  PasswordToggleIcon,
+  AlertMessage,
+  InputLayout
+} from "./Auth";
+import { signIn } from "../../../pages/api/AuthApi";
+import SnsAuth from "./SnsAuth";
+import { useForm } from "react-hook-form";
+
+const Form = () => {
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+    watch,
+    setError
+  } = useForm({
+    mode: "onBlur",
+    defaultValues: {
+      email: "",
+      password: "",
+      passwordConfirm: ""
+    }
+  });
+
+  const onSubmit = async () => {
+    if (!errors.email?.message && !errors.password?.message) {
+      await signIn(
+        watch("email"),
+        watch("password"),
+        (inputName, { message }) =>
+          setError(inputName as "email" | "password", { message })
+      );
+    }
+  };
+
+  return (
+    <FormBox onSubmit={handleSubmit(onSubmit)}>
+      <EmailBox isError={!!errors?.email}>
+        <label>이메일</label>
+        <input
+          placeholder="이메일을 입력해 주세요"
+          {...register("email", {
+            required: "이메일을 입력해주세요.",
+            pattern: {
+              value: /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/i,
+              message: "올바른 이메일 주소가 아닙니다."
+            }
```

Comment:
이메일 검증은 이 곳 말고도 다양한 곳에서 사용될 수 있을 거같아요.
'정규식' 또는 더 좋게는 '정규식으로 검사하는 함수' 를 분리해두면 가져다 쓰면 되겠죠
```suggestion
            validate: {
              email: value => isEmail(value) || "올바른 이메일 주소가 아닙니다.",
            }
```

Reply:
그러한 검증함수을 유틸리티로 사용하고 싶은 사람들이 만든 유서깊은 패키지 validator.js를 참고하셔도 좋겠습니다